src/NuGet.Core/NuGet.PackageManagement/Resolution/PrunePackageTree.cs

```diff
@@ -16,6 +16,29 @@ namespace NuGet.PackageManagement
     public static class PrunePackageTree
     {
         /// <summary>
+        /// Remove some of the prerelease packages in update scenarios
+        /// </summary>
+        public static IEnumerable<SourcePackageDependencyInfo> PrunePrereleaseExceptAllowed(IEnumerable<SourcePackageDependencyInfo> packages, IEnumerable<PackageIdentity> installedPackages, bool isUpdateAll)
+        {
+            var allowedPackageIdentity = new HashSet<PackageIdentity>(installedPackages.Where(p => p.HasVersion && p.Version.IsPrerelease), PackageIdentityComparer.Default);
+
+            if (isUpdateAll)
+            {
+                // If this is an Update All scenario then we will allow package that are already prerelease to pick any other prerelease alternatievs
+
+                var allowedPackageId = new HashSet<string>(allowedPackageIdentity.Select(p => p.Id), StringComparer.OrdinalIgnoreCase);
```

Comment:
This should be able to use a hashset of all installed packages, not just prerelease.